### DISCUSSION TITLE
Should we release 0.73.3?

### DISCUSSION THREAD

**Discussion by fortmarek** [reactwg/react-native-releases#98] [Patches]
Current Release: [0.73.2](https://github.com/facebook/react-native/releases/tag/v0.73.2)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

**Comment by gituser8796** [comment]
Could you please include an important performance fix for Hermes JSON.stringify?

https://github.com/facebook/hermes/commit/8fd20f29af83ab59c5e21d7de8212bcc904dcc4a

**Reply by blakef** [reply]
@gituser8796 we don't normally pick performance fixes in a patch release (unless the release introduced the issue).  I can see that @neildhar suggested picking this change for an [earlier release](https://github.com/facebook/hermes/issues/1008#issuecomment-1663083646), so I think it's ok to do.

**Reply by blakef** [reply]
✅ Picked @ `hermes-2024-01-25-RNv0.73.3-57e3665290d5e97eaea63d27cb370ce7862a10a6`

**Reply by blakef** [reply]
This is [failing on Hermes CI](https://app.circleci.com/pipelines/github/facebook/hermes/5499/workflows/8e4f4dee-0ba6-425c-9b39-9574c2578374/jobs/55292). ~I'll see if we can resolve this, otherwise I'll put it from the cut.~ Removed this from 0.73.3.

**Reply by neildhar** [reply]
@blakef That failure looks unrelated. Hermes CI pulls RN from trunk rather than the RN version associated with its branch, and it's failing because we added a new method to JSI, which is present in RN trunk, but not in the Hermes branch for RN 0.73.

(I'm not advocating either way on picking the fix, just noting that the failure is unrelated)

**Reply by blakef** [reply]
Thanks for clearing that up Neil!

**Reply by efstathiosntonas** [reply]
@blakef just saw the 0.73.3 changelog and it's not there (haven't gone through the commits). Is this included or not?. It's not clear what you decided 😄. Thanks

**Reply by blakef** [reply]
Hi @efstathiosntonas, I've reread the comments above and can see how that's not clear.  Apologies, you've been very patient.  Let me talk to the rest of the release crew and see how to get this (and the CI issue we hit) resolved in the next patch release.

**Reply by efstathiosntonas** [reply]
@blakef no worries! thanks to everyone for your hard work.

**Reply by lunaleaps** [reply]
Following up on this, this commit is on the static hermes branch. I'm not familiar with the status of that branch and its distinction from main but I'd lean on not picking from a non-main branch as future releases also won't have this change. cc @neildhar

**Reply by blakef** [reply]
@neildhar is there anyway we could specify the commit of react native when kicking off Hermes' CI?

**Reply by neildhar** [reply]
@lunaleaps Good catch, it looks like we only made this change in Static Hermes, given that, I'd avoid picking it.

@blakef Our CI currently just pulls from trunk. It's possible to extend it to allow setting a specific RN version, although the e2e test that pulls RN is very simple, and is unlikely to add any value over the testing done on the RN side.

**Reply by gituser8796** [reply]
@neildhar would it be possible to get this on the main branch? We notice a significant performance hit on our app due to this issue. Any help here would be appreciated!

**Reply by neildhar** [reply]
@gituser8796 This seems reasonable to put on our main branch, we'll port it over.

**Reply by gituser8796** [reply]
Thx @neildhar! How do we get it into 0.73.5? @lunaleaps

**Reply by lunaleaps** [reply]
@gituser8796 When the commit is on main, please make a pick request on the latest stable.

**Comment by wfern** [comment]
To fix a release build error due to a casing issue in hermes tarball path

https://github.com/facebook/react-native/commit/2e2f8a668907552070a2a53a47137b1449b66bbd

**Reply by blakef** [reply]
✅ Picked

**Comment by cipolleschi** [comment]
Fix for `symbol not found _jump_fcontext with use_frameworks!` (#39945):
* https://github.com/facebook/react-native/pull/42227

**Reply by blakef** [reply]
✅ Picked @ merge.

**Comment by tido64** [comment]
This is a fix needed for pnpm setups: https://github.com/facebook/react-native/commit/9155e2d1828e04d6144de80bb543d50936500377

**Reply by blakef** [reply]
✅ Picked

**Comment by Crhaj** [comment]
Could we please consider these 2 fixes which should resolve long term issue with Touchable* stopping to fire onPress events?

- [facebook/react-native@e4708d6](https://github.com/facebook/react-native/commit/e4708d661bf3d01ec857635f04a4aabf9d954a5e)
- [facebook/react-native@5416634](https://github.com/facebook/react-native/commit/54166342f01fc74236ba6146a4c0f604189017e4)

**Reply by skinsapp** [reply]
Would definitely appreciate this as well.

**Reply by blakef** [reply]
✅ Picked

**Reply by GorbunovV** [reply]
Picked

**Comment by idrissakhi** [comment]
Would you please include this fix to resolve Image resolving issue on tests (Jest tests)

* https://github.com/facebook/react-native/pull/41957

**Reply by blakef** [reply]
This change hasn't been accepted at the time we're cutting.

**Comment by szymonrybczak** [comment]
CLI Update - fixes running `run-android` command on Windows, and replacing `packageName` in new Kotlin template: https://github.com/facebook/react-native/pull/42279

**Reply by blakef** [reply]
✅ Picked @ merge by @huntie.

**Comment by cipolleschi** [comment]
Please pick https://github.com/facebook/react-native/commit/63e893d23d84c9bf65bad64fc359b9686eb19b4d to fix an Xcode issue

**Reply by blakef** [reply]
✅ Picked

**Comment by cipolleschi** [comment]
Please pick https://github.com/facebook/react-native/pull/42382
This fixes the podfile.lock and resolve a deadlock that prevent the apps from working in Release mode when the new architecture is enabled.

**Reply by migueldaipre** [reply]
Maybe fix [this issue](https://github.com/software-mansion/react-native-reanimated/issues/5497)?

**Reply by cipolleschi** [reply]
No, unfortunately. I tried with the reproducer @gabrieldonadel provided and that problem is still happening.
The two might be related, though, and this fix is the first part of the fix for that problem too, but I need to investigate more to understand how to fix that crash.

**Reply by cipolleschi** [reply]
but [this PR](https://github.com/facebook/react-native/pull/42396) that I just opened, should @migueldaipre.
Could you try to pick it and give it a shot?

It for sure fixes the crash, but I'm not sure about the expected behavior in a complex scenario.

The technical reasons is that:
- a refresh is invalidating the bridge
- an event is sent
- RCTDeviceInfo (which was not invalidated) reacts to the event
- being the bridge invalidated, some pieces are still missing ==> crash.

With the fix, we properly invalidate the `RCTDeviceInfo`, but it could be that by doing that we drop some events.

If you could try the patch in your repro and can report whether everything seems to look fine or not, it would be greatly appreciated!

_Note: the PR is against main. If it fixes, we might be able to cherry pick the fix in 0.73.3_

cc. @gabrieldonadel, you might be interested in trying out the fix as well. And infinite thanks for the repro! 🙏

**Reply by migueldaipre** [reply]
For me didn't work and the same crash still occurs. 
I also tested both PRs together, with no news

Should I do more than just create a patch package and run the application?

**Reply by cipolleschi** [reply]
@migueldaipre thanks for the report. In theory a patch-package should be enough. The PR fixes the behavior reported by Gabriel. Could it be that your problem is different? It might be that we have other Modules that are not invalidated as expected and a colleague is investigating that.

If you can provide a reproducer starting from [this repo](https://github.com/react-native-community/reproducer-react-native) it would speed up our work dramatically.

**Reply by gabrieldonadel** [reply]
Thanks for fixing this @cipolleschi, I just tested your PR and now everything is working as expected

**Reply by blakef** [reply]
✅ Picked @ merge.

**Comment by matinzd** [comment]
Please pick https://github.com/facebook/react-native/commit/96ed1190c624869af96e464b009e0c0234076893

This fixes the issue regarding the permission contracts on Android 14.  Currently it’s blocking https://github.com/matinzd/react-native-health-connect/pull/60 and anything related to this matter.

**Reply by blakef** [reply]
✅ Picked

**Comment by dmytrorykun** [comment]
https://github.com/facebook/react-native/pull/40860 to address https://github.com/facebook/react-native/issues/42477

**Reply by blakef** [reply]
✅ Picked

**Comment by cipolleschi** [comment]
We should pick https://github.com/facebook/react-native/commit/d46d80d2ef6cdc05c5a1b2429ed1df93410ae36f to fix an issue with reload.

**Reply by blakef** [reply]
✅ Picked

**Comment by gabrieldonadel** [comment]
Can we pick https://github.com/facebook/react-native/pull/42637 when it gets merged? This will fix the unexpected changes on Info.plist when building on CI

**Reply by gabrieldonadel** [reply]
This is the commit https://github.com/facebook/react-native/commit/41c601e4786b544fdd1fca138b0e0c61dbb8eba2

**Reply by blakef** [reply]
✅ Picked

**Comment by blakef** [comment]
Request picking facebook/react-native#42306 which resolves a crash.

**Reply by blakef** [reply]
✅ Picked

**Comment by cipolleschi** [comment]
[This commit](https://github.com/facebook/react-native/commit/b1e92d6e0d3a2b27cbda8ad5d1fdbebcdec9a70b) is pretty important and was lying in main since October.
Can we pick it in 0.73.3? :pray:

**Reply by blakef** [reply]
✅ Picked

**Comment by blakef** [comment]
We should pick facebook/react-native@3869ae4d80d1da8c4e3fe7449f2926c5e1575beb when it lands to fix a Cocoapods issue.

**Reply by blakef** [reply]
✅ Picked

**Reply by efstathiosntonas** [reply]
@blakef any idea when 0.73.3 will be released? thanks

**Reply by blakef** [reply]
We're just testing everything now (because of the late picks), if we don't find any issues it should happen reasonably quickly.  Can't be more specific than that.

**Comment by ansh** [comment]
Will Expo publish `0.73.3` soon?
